versions.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+versions=( "$@" )
+if [ ${#versions[@]} -eq 0 ]; then
+	versions=( */ )
+	json='{}'
+else
+	json="$(< versions.json)"
+fi
+versions=( "${versions[@]%/}" )
+
+allVersions="$(
+	git ls-remote --tags https://github.com/TryGhost/Ghost.git \
+		| sed -rne 's!^.*\trefs/tags/v?|\^\{\}$!!g; /^[0-9][.][0-9]+/p' \
+		| sort -ruV
+)"
+
+cliVersion="$(
+	git ls-remote --tags https://github.com/TryGhost/Ghost-CLI.git \
+		| sed -rne 's!^.*\trefs/tags/v?|\^\{\}$!!g; /^[0-9][.][0-9]+/p' \
+		| grep -vE -- '-(alpha|beta|rc)' \
+		| sort -ruV \
+		| head -n1
+)"
+
+for version in "${versions[@]}"; do
+	rcVersion="${version%-rc}"
+	rcGrepV='-v'
+	if [ "$rcVersion" != "$version" ]; then
+		rcGrepV=
+	fi
+	rcGrepV+=' -E'
+	rcGrepExpr='alpha|beta|rc'
+	export version
+
+	fullVersion="$(
+		echo "$allVersions" \
+			| grep -E "^${rcVersion}([.-]|$)" \
+			| grep $rcGrepV -- "$rcGrepExpr" \
+			| head -1
+	)"
+	if [ -z "$fullVersion" ]; then
+		echo >&2 "error: cannot determine full version for '$version'"
+		exit 1
+	fi
+
+	# get a list of architectures supported by the sharp module's prebuilt libraries
+	# we cannot build it on other arches since the dep, libvips, is usually too old in Debian and Alpine
+	doc="$(curl -fsSL "https://raw.githubusercontent.com/TryGhost/Ghost/refs/tags/v$fullVersion/yarn.lock" \
+		| jq --compact-output --raw-input --null-input '
+			reduce (
+				inputs
+				| capture("^ *\"@img/sharp-(?<name>linux[a-z0-9-]+)\" \"[0-9.]+\"$")
+				| .name
+				| split("-")
```

Comment:
Given the immediate `split` to take off more data, couldn't this instead massage the regex a little so that `.name` is the exact value we need?

```suggestion
				| capture("^ *\"@img/sharp-(?<dist>linux[a-z]*)-(?<arch>[a-z0-9]+)\" \"[0-9.]+\"$")
```

(and then using `.dist` and `.arch` instead of `.[0]` and `.[1]` below - could even be `.dist |= { ... }[.] | .arch |= { ... }[.]`)